File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLContact.java

package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.TLContext;
import com.github.badoualy.telegram.tl.core.TLObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readInt;
import static com.github.badoualy.telegram.tl.StreamUtils.readTLBool;
import static com.github.badoualy.telegram.tl.StreamUtils.writeInt;
import static com.github.badoualy.telegram.tl.StreamUtils.writeTLBool;


public class TLContact extends TLObject {

    public static final int CLASS_ID = 0xf911c994;

    public TLContact() {

    }


    public TLContact(        int _userId,         boolean _mutual) {
        this.userId = _userId;
        this.mutual = _mutual;

    }


    public int getClassId() {
        return CLASS_ID;
    }


    protected int userId;

    protected boolean mutual;


    public int getUserId() {
        return userId;
    }

    public void setUserId(int value) {
        this.userId = value;
    }

    public boolean getMutual() {
        return mutual;
    }

    public void setMutual(boolean value) {
        this.mutual = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeInt(this.userId, stream);
        writeTLBool(this.mutual, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.userId = readInt(stream);
        this.mutual = readTLBool(stream);
    }


    @Override
    public String toString() {
        return "contact#f911c994";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLContact.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.core.TLObject
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException

/**
 * contact#f911c994
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLContact() : TLObject() {
    var userId: Int = 0

    var mutual: Boolean = false

    private val _constructor: String = "contact#f911c994"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(userId: Int, mutual: Boolean) : this() {
        this.userId = userId
        this.mutual = mutual
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeInt(userId)
        writeBoolean(mutual)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        userId = readInt()
        mutual = readBoolean()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += SIZE_INT32
        size += SIZE_BOOLEAN
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLContact) return false
        if (other === this) return true

        return userId == other.userId
                && mutual == other.mutual
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xf911c994.toInt()
    }
}
